components/brave_wallet/browser/BUILD.gn

```diff
@@ -164,14 +164,16 @@ static_library("browser") {
       "brave_wallet_pin_service.h",
     ]
   }
-  deps = [
+  public_deps = [
     ":constants",
     ":generated_json_rpc_responses",
     ":generated_swap_responses",
     ":hd_keyring",
     ":pref_names",
     ":transaction",
     ":utils",
+  ]
+  deps = [
```

Comment:
Why is this needed? If a target wants to depend on ex. `constants`, it will put `//brave/components/brave_wallet/browser:constants` into its dep instead of `//brave/components/brave_wallet/browser` to get all public deps implicitly.

Reply:
>That's correct, but do we really want to micro-manage all these sub-dependencies when we add //brave/components/brave_wallet/browser dependency?

I don't think `public_deps` is designed for eliminating these sub deps, it is used for ex. target A depends on `ui` which depends on `//brave/components/brave_wallet/browser` and A couldn't depends on `//brave/components/brave_wallet/browser:constants` due to visibility but want to include headers from it.
It's usually used for `components`, `static_library`, `shard_library` or some targets that have visibility restriction.

Reply:
> It's usually used for `components`, `static_library`, `shard_library` or some targets that have visibility restriction.

I'd like to point out another `public_deps` use case for anyone reading this thread: when a target uses generated headers in its own public headers, such as `mojom-forward.h`, we should use `public_deps += [ ":mojom" ]`, otherwise the build might fail occasionally if an upstream target doesn't depend on `:mojom` explicitly.

So `public_deps` is not only about visibility, but also about propagating the correct build graph dependency to upstream targets so ninja knows not to build some files until some dependencies are ready.